.github/workflows/config.yml

```diff
@@ -0,0 +1,735 @@
+        path: 'celo-monorepo/${{ env.CONTRACTS_BUILD_PATH }}'
+
+  go-modules:
+    name: Setup go
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Setup go
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ env.GO_VERSION }}
+        check-latest: false
+        cache: true
+
+  check-imports:
+    name: Check imports
+    needs: go-modules
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Setup golang
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ env.GO_VERSION }}
+        check-latest: false
+        cache: true
+    - name: Check imports to ensure we are using github.com/celo-org/celo-blockchain
+      run: ./scripts/check_imports.sh
+
+  lint:
+    name: Lint code
+    runs-on: ubuntu-latest
+    needs: go-modules
+
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Setup golang
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ env.GO_VERSION }}
+        check-latest: false
+        cache: true
+    - name: Run linter
+      run: |
+        # so that the tee call at the end doesn't hide the real exit code
+        set -o pipefail
+        mkdir -p /tmp/test-results
+        go run build/ci.go ensure-linter
+        ./build/cache/golangci-lint-1.42.0-linux-amd64/golangci-lint run --config .golangci.yml --out-format junit-xml ./... | tee /tmp/test-results/go-lint-report.xml
+    - name: Upload linter report
+      uses: actions/upload-artifact@v3
+      with:
+        name: linter-report
+        path: /tmp/test-results
+
+  unit-tests:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    needs:
+    - go-modules
+    - prepare-system-contracts
+
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Setup golang
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ env.GO_VERSION }}
+        check-latest: false
+        cache: true
+    - name: Restore system contracts
+      uses: actions/download-artifact@v3
+      with:
+        name: system-contracts
+        path: ${{ env.SYSTEM_CONTRACTS_PATH }}
+    - name: Prepare ethersjs project
+      run: make prepare-ethersjs-project
+    - name: Install JUnit report
+      run: go get github.com/jstemmer/go-junit-report
+    - name: Run tests
+      run: |
+        mkdir -p /tmp/test-results
+        trap "go-junit-report < /tmp/test-results/go-test.out > /tmp/test-results/go-test-report.xml" EXIT
+        CI=true go test -v -cover -coverprofile=coverage.out ./... | tee /tmp/test-results/go-test.out
+    # TODO: figure out if we need a token
+    - name: Upload coverage to codecov
+      uses: codecov/codecov-action@v3
+    - name: Upload test report
+      uses: actions/upload-artifact@v3
+      with:
+        name: unit-test-report
+        path: /tmp/test-results
+
+  publish-test-results:
```

Comment:
nit: does it make sense to have an own job for this?

Reply:
Agreed, removed.